rdmd_test.d

```diff
@@ -46,40 +50,55 @@ bool verbose = false;
 
 void main(string[] args)
 {
-    string buildCompiler = "dmd";  // e.g. dmd/gdmd/ldmd
-    string rdmd = "rdmd.d";
+    string rdmd; // path to rdmd executable
+    string defaultCompiler; // name of default compiler expected by rdmd
     bool concurrencyTest;
     string model = "64"; // build architecture for dmd
     string testCompilerList; // e.g. "ldmd2,gdmd" (comma-separated list of compiler names)
 
-    getopt(args,
-        "compiler", &buildCompiler,
-        "rdmd", &rdmd,
-        "concurrency", &concurrencyTest,
-        "m|model", &model,
-        "test-compilers", &testCompilerList,
-        "v|verbose", &verbose,
+    auto helpInfo = getopt(args,
+        "rdmd", "[REQUIRED] path to rdmd executable to test", &rdmd,
+        "rdmd-default-compiler", "[REQUIRED] default D compiler used by rdmd executable", &defaultCompiler,
+        "concurrency", "whether to perform the concurrency test cases", &concurrencyTest,
+        "m|model", "architecture to run the tests for [32 or 64]", &model,
+        "test-compilers", "comma-separated list of D compilers to test with rdmd", &testCompilerList,
+        "v|verbose", "verbose output", &verbose,
     );
 
-    enforce(rdmd.exists, "Path to rdmd does not exist: %s".format(rdmd));
+    void reportHelp(string errorMsg = null, string file = __FILE__, size_t line = __LINE__)
+    {
+        defaultGetoptPrinter("rdmd_test: a test suite for rdmd\n\n" ~
+                             "USAGE:\trdmd_test [OPTIONS]\n",
+                             helpInfo.options);
+        enforce(errorMsg is null, errorMsg, file, line);
+    }
 
-    // path/to/rdmd.exe (once built)
-    string rdmdApp = tempDir().buildPath("rdmd_app_") ~ binExt;
-    if (rdmdApp.exists) std.file.remove(rdmdApp);
+    if (helpInfo.helpWanted)
+    {
+        reportHelp();
+        return;
+    }
 
-    // compiler needs to be an absolute path because we change directories
-    if (buildCompiler.canFind!isDirSeparator)
-        buildCompiler = buildNormalizedPath(buildCompiler.absolutePath());
+    if (rdmd.length == 0)
+        reportHelp("ERROR: missing required --rdmd flag");
 
-    auto res = execute([buildCompiler, modelSwitch(model), "-of" ~ rdmdApp, rdmd]);
+    if (defaultCompiler.length == 0)
+        reportHelp("ERROR: missing required --rdmd-default-compiler flag");
```

Comment:
getopt supports this out of the box.

Reply:
I'm aware of `std.getopt.config.required`, but I found the behaviour really undesirable (it was very unhelpful in its error messages).  So it seemed better to do something small, manual and actually helpful in its output.

Reply:
Hmm, I guess we have to improve std.getopt then at some point...